### PR TITLE
Add ability to cancel a distirbuted job with a local streams install

### PR DESCRIPTION
**Pull request by ddebrunner**

### CODE REVIEW

File: python/packages/streamsx/st.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    args = ['canceljob']
+    if force:
+        args.append('--force')
+    args.append(str(job_id))
+
+    return _run_st(args)
+
+def _run_st(args, lines=None):
+    args.insert(0, os.path.join(_install, 'bin', 'streamtool'))
+    process = subprocess.Popen(args, stdout=subprocess.PIPE, stdin=subprocess.DEVNULL)
```
Comment (wmarshall484) [inline python/packages/streamsx/st.py]: Should we also redirect stderr? Not sure about the cases in which streamtool writes to stderr.

Comment (ddebrunner) [inline python/packages/streamsx/st.py]: Where to?

I was assuming that it went to the callers stderr.

_With the default settings of None, no redirection will occur; the child’s file handles will be inherited from the parent._

File: python/packages/streamsx/rest.py [hunk 2]
```diff
@@ -48,23 +49,38 @@ def __init__(self, username=None, password=None, resource_url=None, config=None)
...
 
             # Obtain the streams SWS REST URL
             rest_api_url = VcapUtils.get_rest_api_url_from_creds(credentials)
 
             # Create rest connection to remote Bluemix SWS
             self.rest_client = StreamsRestClient(credentials['userid'], credentials['password'], rest_api_url)
             self.resource_url = rest_api_url
+
+        elif username and password and st._has_local_install:
+            self.resource_url = st.get_rest_api()
```
Comment (wmarshall484) [inline python/packages/streamsx/rest.py]: Shouldn't we ensure that the `resource_url` provided in the call to `__init__` and the `resource_url` returned from `st.get_rest_api()` are the same? We don't want to end up cancelling jobs from Streams install A when we meant to cancel jobs from Streams install B

Comment (ddebrunner) [inline python/packages/streamsx/rest.py]: There is no `resource_url` in this case.

**Pull request merged**